src/plugins/dataflow-tool/components/ui/dataflow-program-topbar.tsx

```diff
@@ -1,137 +1,131 @@
-  duration: string;
-  width: number;
-  hours: string;
-  minutes: string;
-  seconds: string;
-}
-const CountdownTimerComponent: React.SFC<CountdownTimerProps> = (props: CountdownTimerProps) => {
-  return (
-    <div className="running-container countdown">
-      <div className="total">
-        {`Duration: ${props.duration}`}
-      </div>
-      <div className="remaining">
-        <div className="progress-bar" style={{width: props.width.toString() + "px"}}/>
-        <div className="progress-time">
-        {`${props.hours}:${props.minutes}:${props.seconds}`}
-        </div>
-      </div>
-    </div>
-  );
-};
+// interface CountdownTimerProps {
+//   duration: string;
+//   width: number;
+//   hours: string;
+//   minutes: string;
+//   seconds: string;
+// }
+// const CountdownTimerComponent: React.SFC<CountdownTimerProps> = (props: CountdownTimerProps) => {
+//   return (
+//     <div className="running-container countdown">
+//       <div className="total">
+//         {`Duration: ${props.duration}`}
+//       </div>
+//       <div className="remaining">
+//         <div className="progress-bar" style={{width: props.width.toString() + "px"}}/>
+//         <div className="progress-time">
+//         {`${props.hours}:${props.minutes}:${props.seconds}`}
+//         </div>
+//       </div>
+//     </div>
+//   );
+// };
 
-interface DurationSelectorProps {
-  onRunProgramClick: () => void;
-  programRunTimes: ProgramRunTime[];
-  programDefaultRunTime: number;
-  onProgramTimeSelectClick: (type: number) => void;
-  isRunEnabled: boolean;
+interface RateSelectorProps {
+  rateOptions: ProgramDataRate[];
+  dataRate: number;
+  onRateSelectClick: (rate: number) => void;
   readOnly: boolean;
 }
 
-const DurationSelectorComponent: React.SFC<DurationSelectorProps> = (props: DurationSelectorProps) => {
+const RateSelectorComponent: React.FC<RateSelectorProps> = (props: RateSelectorProps) => {
   const handleSelectChange = (event: React.ChangeEvent<HTMLSelectElement>) => {
-    props.onProgramTimeSelectClick(Number(event.target.value));
+    props.onRateSelectClick(Number(event.target.value));
   };
   return (
     <div className="running-container">
-      <div className="duration" title="Set Program Duration">
+      <div className="datarate" title="Set Program Data Rate">
         <div className="label-back">
-          <div className="label">Duration</div>
+          <label className="label" htmlFor="rate-select">Data Rate</label>
         </div>
-        <div className="duration-options-back">
-          <div className="duration-options">
+        <div className="datarate-options-back">
+          <div className="datarate-options">
             <select onChange={handleSelectChange}
-              disabled={!props.isRunEnabled || props.readOnly}
-              value={props.programDefaultRunTime.toString()}
+              disabled={props.readOnly}
+              value={props.dataRate.toString()}
+              id="rate-select"
```

Comment:
Not sure what the best solution is here, but given that it is possible to have two Dataflow tile instances in a document, which would result in duplicate DOM ids, perhaps there should be a `TODO` comment nearby to consider disambiguating ids in that situation. Given how early we are in the development process and the fact that multiple Dataflow tiles isn't an _expected_ configuration, I don't think it needs to be addressed in this PR.

Reply:
Good call, I hadn't considered a collision between multiple tiles here. I'll put in the `TODO` and maybe address this when we add titles to the DF tiles.